lib/matchSourceExpression.js

```diff
@@ -0,0 +1,180 @@
+var urlModule = require('url');
```

Comment:
This entire file has no Assetgraph dependencies and the code seems well documented enough to proudly live on its own as a CSP tooling module on npm